### ISSUE TITLE
ReaderWidget does not scan anything on iPad

### ISSUE THREAD

**Issue by kaboc** [khoren93/flutter_zxing#24]
I'm really impressed that it works really stable on Android. I don't have any issues that I had with other similar packages. It's great.

However, I tried it next on my iPad and found that the same code wouldn't work. Are you aware of the issue that `onScan` is never called on iPadOS? I have no idea if the same thing also happens on iOS because I don't have an iPhone.

It appears that it's only an issue of scanning.

- Camera preview is working.
- The `onControllerCreated` callback is called.
- `scannerOverlay` has no issue either.

Just FYI, a message "setFlashModeFailed: Devide does not have flash capabilities" is printed in the console when the camera is ready, although I don't think this has anything to do with the issue.

**Comment by khoren93** [comment]
Hi @kaboc 
Thank you for an issue report.
I did not have such a problem, I tested on iPad Pro 11 and iPad Pro 9,7 and everything works fine.
Can you please try scanning from an image and tell me if it works?

**Comment by kaboc** [comment]
The same QR code as the one that wasn't scanned with ReaderWidget was successfully scanned using the `processCameraImage()` method in `startImageStream()` on the CameraController obtained with `onControllerCreated`.

Another QR code was scanned with `readBarcodeImageUrl()` too.

Hope this gives you a clue.

**Comment by kaboc** [comment]
My environment

- Flutter: 3.0.5
- iPad: 8th gen (OS: 15.5)

**Comment by LinHaoLove** [comment]
Yes, I have the same problem.
Flutter: 3.0.5
iPad Air ==>iOS 12.5.5 (16H62)
iPad mini 3 ==>iOS 12.5.5 (16H62)
iPad mini 5 ==>iOS 15.6 (19G71)

I just only use the ReaderWidget.
I don't know how to refer to CameraController.

**Comment by khoren93** [comment]
@LinHaoLove to refer to CameraController you can use onControllerCreated property like this
```dart
ReaderWidget(
    onScan: (CodeResult result) {
      // do something with the result
    },
    onControllerCreated: (controller) {
      // do something with the controller
    },
)
```

**Comment by khoren93** [comment]
Fixed in v1.0.1
